sdk/eventhub/event-hubs/samples/expressSample/src/eventProducer.ts

```diff
@@ -33,32 +30,34 @@ export class EventProducer {
   }
 
   public async send(requestId: string, payload: any) {
-    if (this.batch === undefined) {
-      this.batch = await this.lock.acquire(this.lockKey, () => {
-        if (this.batch === undefined) {
-          console.log(`Acquire lock ${this.lockKey} and create a new batch`);
-          this.lastBatchCreatedTime = Date.now();
-          return this.producerClient.createBatch();
-        } else {
-          console.log(
-            `Acquire lock ${this.lockKey} and skip create. this.batch count: ${this.batch.count}`
-          );
-          return this.batch;
-        }
-      });
-    }
-    const isAdded = this.batch.tryAdd(this.constructEventData(requestId, payload));
+    const batch = await this.getOrCreateBatch();
+    const eventData = this.constructEventData(requestId, payload)
+    const isAdded = batch.tryAdd(eventData);
 
-    if (!isAdded || this.batch.count >= this.batchSendNumber) {
-      const curBatch = this.batch;
-      this.batch = undefined;
-      await this.sendBatch(curBatch);
+    if (!isAdded || batch.count >= this.batchSendNumber) {
+      this.awaitableBatch = undefined;
+      await this.sendBatch(batch);
       if (!isAdded) {
-        console.log(`Up to batch size limit, add event failed. requestId: ${requestId}`);
+        const batch = await this.getOrCreateBatch();
```

Comment:
This will cause a risk conditions.  We do need a lock here.  As we know, `await` only block the current eventloop.  But the send function can happen in different event loop, each request is an eventloop. Thinking about this scenario: There are two coming requests at the same time, the first one create a batch add a item, but the second one also create a new one, it will cause overwrite batch and lose data.